bofire/data_models/strategies/enting.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any, Dict, Literal, Type
+
+from bofire.data_models.constraints.api import (
+    Constraint,
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+)
+from bofire.data_models.features.api import (
+    CategoricalDescriptorInput,
+    CategoricalInput,
+    ContinuousInput,
+    ContinuousOutput,
+    DiscreteInput,
+    Feature,
+)
+from bofire.data_models.objectives.api import MinimizeObjective, Objective
+from bofire.data_models.strategies.strategy import Strategy
+
+
+class EntingStrategy(Strategy):
+    type: Literal["EntingStrategy"] = "EntingStrategy"
+    enting_params: Dict[str, Any] = {}
```

Comment:
As we have also the XGBSurrogate in BoFire, I am wondering, if one could export the hyperparams of the XGB surrogates that are used to the user as we are also doint it for the botorch based strategies. You can have a look here: https://github.com/experimental-design/bofire/blob/a4ca3eb2ecf8348e5c387cbef273191c1ba54d45/bofire/data_models/strategies/predictives/botorch.py#L35 and here: https://github.com/experimental-design/bofire/blob/main/bofire/data_models/surrogates/xgb.py

Reply:
I agree for the `enting_params`, and I will create an EntingSurrogate to include the parameters (these are unique to ENTMOOT and I'll have to manually collate them).

The `solver_params` attribute is a little trickier - this is passed directly to the solver, and so is solver agnostic. Since we don't have a list of every solver, and every associated set of parameters, I think this might be trickier to type hint. A solution is to just not allow passing anything to the solver, or use a default set of options, but you then lose some freedom. This mostly comes down to ENTMOOT using Pyomo where BoFire uses ipopt: losing freedom to use arbitrary solvers might not be such a bad thing.

Reply:
> I am not a big fan of having dictionary fields that can have anything inside, as it is somehow contraty of using a validation framework like pydantic. Can you provide the possible parameters and its defaults and types and validation, that can be used as fields of the data model?

This change should be made directly in entmoot (@spiralulam wanted to do it); Johannes's argument is in general a good one and not just valid here